numpy/lib/tests/test_shape_base.py

```diff
@@ -27,6 +27,27 @@ def test_3d(self):
         assert_array_equal(apply_along_axis(np.sum, 0, a),
                            [[27, 30, 33], [36, 39, 42], [45, 48, 51]])
 
+    def test_preserve_subclass(self):
+        m = np.matrix(np.ones((4, 3)))
+        result = apply_along_axis(np.sum, 0, m)
+        assert isinstance(result, np.matrix)
+        assert_array_equal(
+            result, np.matrix([4, 4, 4])
+        )
+
+    def test_subclass(self):
+        class MinimalSubclass(np.ndarray):
+            data = 1
+
+        def minimal_function(array):
+            return array.data
```

Comment:
Again, for this minimal function where the output is not itself an `ndarray` subclass, I don't quite see why the result should be an `ndarray` subclass.